src/main/java/com/iexec/core/security/ChallengeService.java

```diff
@@ -16,30 +16,23 @@
 
 package com.iexec.core.security;
 
-import net.jodah.expiringmap.ExpirationPolicy;
-import net.jodah.expiringmap.ExpiringMap;
 import org.apache.commons.lang3.RandomStringUtils;
 import org.springframework.stereotype.Service;
 
-import java.util.concurrent.TimeUnit;
+import java.util.concurrent.ConcurrentHashMap;
 
 @Service
 public class ChallengeService {
 
-    // Map <WorkerWalletAdress, Challenge>
-    // this map will automatically delete entries older than one hour, ExpiringMap is thread-safe
-    private final ExpiringMap<String, String> challengeMap;
+    private final ConcurrentHashMap<String, String> challengesMap = new ConcurrentHashMap<>();
 
-    ChallengeService() {
-        this.challengeMap = ExpiringMap.builder()
-                .expiration(60, TimeUnit.MINUTES)
-                .expirationPolicy(ExpirationPolicy.CREATED)
-                .build();
+    public String getChallenge(String workerWallet) {
+        challengesMap.computeIfAbsent(workerWallet, wallet -> RandomStringUtils.randomAlphabetic(10));
+        return challengesMap.get(workerWallet);
```

Comment:
```suggestion
        return challengesMap.computeIfAbsent(workerWallet, wallet -> RandomStringUtils.randomAlphabetic(10));
```
`computeIfAbsent` returns the current or newly-computed value